### PR TITLE
fix(2069): check disallowed chars

### LINKED ISSUE #2069: disallow some characters for `custom character sets`
disallow using of characters according to the [wiki](https://github.com/tempesta-tech/tempesta/wiki/HTTP-security#custom-character-sets) (at the config parsing layer)
Test to reproduce - `t_custom_branges/test_http_uri_brange.TestConfigParsing`

### PR DESCRIPTION
**Pull request by kingluo**
Fix #2069

test (add invalid char 0x10, then cfg parsing is supposed to fail):

```diff
diff --git a/t_custom_branges/test_http_brange.py b/t_custom_branges/test_http_brange.py
index 796df20..b67ffc9 100644
--- a/t_custom_branges/test_http_brange.py
+++ b/t_custom_branges/test_http_brange.py
@@ -200,7 +200,7 @@ tls_match_any_server_name;
                 # Example from issue #2030:
                 # Allow only following characters in URI (no '%'): /a-zA-Z0-9&?:-._=
                 custom_uri_branch=(
-                    "0x2f 0x41-0x5a 0x61-0x7a 0x30-0x39 0x26 0x3f 0x3a 0x2d 0x2e 0x5f 0x3d"
+                    "0x10 0x2f 0x41-0x5a 0x61-0x7a 0x30-0x39 0x26 0x3f 0x3a 0x2d 0x2e 0x5f 0x3d"
                 ),
                 uri="/js/prism.min.js",
                 expected_status="200",
```

```bash
./run_tests.py t_custom_branges.test_http_brange.TestBrangeH2
```

### CODE REVIEW

File: fw/http.c [hunk 1]
```diff
@@ -7425,15 +7425,59 @@ __cfgop_brange_hndl(TfwCfgSpec *cs, TfwCfgEntry *ce, unsigned char *a)
...
+				T_ERR_NL("Cannot parse interval: '%s'\n", val);
+				return -EINVAL;
+			}
+			if (i0 > 255 || i1 > 255) {
+				T_ERR_NL("Too large interval bounds: '%s'\n", val);
+				return -EINVAL;
+			}
+
+			a[i0++] = 1;
+			printk(KERN_INFO "disallow: %s %d\n", str, a[i0-1]);
```
Comment (krizhanovsky) [inline fw/http.c:7449]: Pleae use Tempesta FW's macros. I think this should be `T_DBG2` or `T_DBG3` (please check the description for which levels we use in which cases)

File: fw/http.c [hunk 2]
```diff
@@ -7425,15 +7425,59 @@ __cfgop_brange_hndl(TfwCfgSpec *cs, TfwCfgEntry *ce, unsigned char *a)
...
+			while (i0 <= i1)
+				a[i0++] = 1;
+
+			start = end + 1;
+		}
+		if (*end++ == '\0') {
+			break;
+		}
```
Comment (krizhanovsky) [inline fw/http.c:7457]:
```suggestion
		if (*end++ == '\0')
			break;
```

File: fw/http.c [hunk 3]
```diff
@@ -7425,15 +7425,59 @@ __cfgop_brange_hndl(TfwCfgSpec *cs, TfwCfgEntry *ce, unsigned char *a)
...
 }
 
-#define TFW_HTTP_CFG_CUSTOM_BRANGE(name)				\
+static int
+__cfgop_brange_parse_disallowed(char *str, unsigned char *a)
+{
+	char *start = str, *end = str;
+
+	while (true) {
+		if (*end == ' ' || *end == '\0') {
```
Comment (krizhanovsky) [inline fw/http.c:7434]:
```suggestion
		if (*end != ' ' && *end != '\0')
		        continue;
```
The line width limit in the coding style mostly targets smaller nesting in the code

**Review by enuribekov-tempesta — APPROVED** [review]
LGTM

File: fw/http.c [hunk 4]
```diff
@@ -7445,14 +7491,14 @@ tfw_cfgop_cleanup_brange_##name(TfwCfgSpec *cs)				\
...
-TFW_HTTP_CFG_CUSTOM_BRANGE(token);
-TFW_HTTP_CFG_CUSTOM_BRANGE(qetoken);
-TFW_HTTP_CFG_CUSTOM_BRANGE(nctl);
-TFW_HTTP_CFG_CUSTOM_BRANGE(ctext_vchar);
-TFW_HTTP_CFG_CUSTOM_BRANGE(xff);
-TFW_HTTP_CFG_CUSTOM_BRANGE(cookie);
-TFW_HTTP_CFG_CUSTOM_BRANGE(etag);
+TFW_HTTP_CFG_CUSTOM_BRANGE(uri, "0x00-0x20");
+TFW_HTTP_CFG_CUSTOM_BRANGE(token, "0x00-0x20 0x2c 0x3b");
+TFW_HTTP_CFG_CUSTOM_BRANGE(qetoken, "0x00-0x20 0x2c");
```
Comment (const-t) [inline fw/http.c:7496]: @RomanBelozerov Why do we disallow `0x2c` legit to qetoken? However for token we disallow `,;` but allow potentially "malicious" characters `"\:/`?

RFC qetoken:
>   quoted-string  = DQUOTE *( qdtext / quoted-pair ) DQUOTE
>   qdtext         = HTAB / SP / %x21 / %x23-5B / %x5D-7E / obs-text

Comment (RomanBelozerov) [reply fw/http.c:7496]: after call discussion we decided not to restrict the list of characters . I added a list of characters that break the parser

Comment (kingluo) [reply fw/http.c:7496]: @RomanBelozerov So this issue is canceled?

Comment (const-t) [reply fw/http.c:7496]: @kingluo Please, change `qetoken` disallowed characters accordingly to RFC. `HTAB / SP / %x21 / %x23-5B / %x5D-7E/ obs-text` must not be disallowed.

File: fw/http.c [hunk 5]
```diff
@@ -7445,14 +7491,14 @@ tfw_cfgop_cleanup_brange_##name(TfwCfgSpec *cs)				\
...
-TFW_HTTP_CFG_CUSTOM_BRANGE(ctext_vchar);
-TFW_HTTP_CFG_CUSTOM_BRANGE(xff);
-TFW_HTTP_CFG_CUSTOM_BRANGE(cookie);
-TFW_HTTP_CFG_CUSTOM_BRANGE(etag);
+TFW_HTTP_CFG_CUSTOM_BRANGE(uri, "0x00-0x20");
+TFW_HTTP_CFG_CUSTOM_BRANGE(token, "0x00-0x20 0x2c 0x3b");
+TFW_HTTP_CFG_CUSTOM_BRANGE(qetoken, "0x00-0x08 0x22 0x5c 0x7f");
+TFW_HTTP_CFG_CUSTOM_BRANGE(nctl, "0x00-0x1f");
+TFW_HTTP_CFG_CUSTOM_BRANGE(ctext_vchar, "0x00-0x1f");
+TFW_HTTP_CFG_CUSTOM_BRANGE(xff, "0x00-0x20 0x2c");
```
[the remark this answers is not in the document]
Comment (kingluo) [reply fw/http.c:7499]: It's from our [wiki](https://github.com/tempesta-tech/tempesta/wiki/HTTP-security#custom-character-sets)

> http_xff_brange - X-Forwarded-For Node ID defined by RFC 7239. Disallowed characters - 0x00-0x20, 0x2c

File: fw/http.c [hunk 6]
```diff
@@ -7445,14 +7491,14 @@ tfw_cfgop_cleanup_brange_##name(TfwCfgSpec *cs)				\
...
-TFW_HTTP_CFG_CUSTOM_BRANGE(token);
-TFW_HTTP_CFG_CUSTOM_BRANGE(qetoken);
-TFW_HTTP_CFG_CUSTOM_BRANGE(nctl);
-TFW_HTTP_CFG_CUSTOM_BRANGE(ctext_vchar);
-TFW_HTTP_CFG_CUSTOM_BRANGE(xff);
-TFW_HTTP_CFG_CUSTOM_BRANGE(cookie);
-TFW_HTTP_CFG_CUSTOM_BRANGE(etag);
+TFW_HTTP_CFG_CUSTOM_BRANGE(uri, "0x00-0x20");
+TFW_HTTP_CFG_CUSTOM_BRANGE(token, "0x00-0x20 0x2c 0x3b");
+TFW_HTTP_CFG_CUSTOM_BRANGE(qetoken, "0x00-0x08 0x22 0x5c 0x7f");
```
[the remark this answers is not in the document]
Comment (kingluo) [reply fw/http.c:7496]: 0x09 is SP (space), which is allowed.

See [rfc7230](https://www.rfc-editor.org/rfc/rfc7230#appendix-B):
>    qdtext = HTAB / SP / "!" / %x23-5B ; '#'-'[' / %x5D-7E ; ']'-'~' / obs-text

[the remark this answers is not in the document]
Comment (const-t) [reply fw/http.c:7496]: Sorry. 0x09 - Is HTAB and we must allow, but we need disallow 0xA-0x1f.

**Pull request merged**